src/Confluent.Kafka/Timestamp.cs

```diff
@@ -80,6 +88,15 @@ public static long DateTimeToUnixTimestampMs(DateTime dateTime)
             }
         }
 
+        /// <summary>
+        ///     Convert a unix timestamp to a DateTime value.
+        /// </summary>
+        /// <param name="timestamp">
+        ///     The unix timestamp to convert.
+        /// </param>
+        /// <returns>
+        ///     The DateTime value associated with <paramref name="timestamp"/>
+        /// </returns>
         public static DateTime UnixTimestampMsToDateTime(long timestamp)
             => new DateTime(1970, 1, 1, 0, 0, 0, 0, System.DateTimeKind.Unspecified)
```

Comment:
I corrected the produce method as you pointed out, though this won't come into play just yet as we have disabled all variants of ProduceAsync that allow a timestamp to be explicitly specified in the upcoming release (because of lack of time and there is stuff around this to be tested, including the marshalling to producev which is definitely wrong). Also, we will move away from long? for timestamps because timestamp of 0 is interpreted by librdkafka to mean 'current time', the reason I introduced null in the first place.

Also fixed Unspecified -> Utc. Good point.